src/components/Map/Map.tsx

```diff
@@ -293,8 +290,8 @@ export default function Map(props: MapProps): JSX.Element {
     if (!geoData) {
       return null;
     }
-    const sources = (geoData as any[]).map((geo: any, index) => (
-      <Source type='geojson' key={index} data={geo.data} id={geo.id}>
+    const sources = (geoData as any[]).map((geo: any) => (
+      <Source type='geojson' key={geo.id} data={geo.data} id={geo.id}>
```

Comment:
Setting the key here to be the source id was an important and surprising change. Internally, mapbox associates the react key with the source id, and if the source id for a given key changes (e.g. if the sources are reordered, resulting in their index changing) then mapbox crashes the whole page.